GitHubDashboard/Controllers/QueryController.cs

```diff
@@ -130,9 +130,42 @@ private async Task<IReadOnlyList<Issue>> GetIssuesAsync(string owner, string rep
 
             // This could throw an ApiValidationException if the milestone doesn't exist in the repo.
             // Catch it in the calling function...
-            var issues = await _gitHubClient.Issue.GetAllForRepository(owner, repository, issueRequest);
-            issues = issues.Where(i => i.PullRequest == null).ToList();
-            return issues;
+            var allIssues = await _gitHubClient.Issue.GetAllForRepository(owner, repository, issueRequest);
+            var issues = allIssues.Where(i => i.PullRequest == null);
+
+            // We now need to exclude the milestone
+            if (!string.IsNullOrEmpty(excludedMilestone) && (excludedMilestone != "undefined"))
+            {
+                issues = issues.Where(i => i.Milestone == null || i.Milestone.Title != excludedMilestone);
+            }
+
+            // We now need to exclude all the issues that have labels that should be excluded
+            if (!string.IsNullOrEmpty(excludedLabels) && (excludedLabels != "undefined"))
+            {
+                var filteredIssues = new List<Issue>();
+                var excludedLabelValues = excludedLabels.Split(',');               
+
+                foreach (Issue i in issues)
+                {
+                    bool skip = false;
+                    foreach (Label l in i.Labels)
+                    {
+                        if (excludedLabelValues.Contains(l.Name))
+                        {
+                            skip = true;
+                        }
+                    }
+
+                    if (!skip)
+                    {
+                        filteredIssues.Add(i);
+                    }
+                }
+
+                issues = filteredIssues;
+            }
+        
+            return issues.ToList();
```

Comment:
Nit: in the case you enter the `if` above, filtered issues is already a `List<T>`.  Consider changing the end of this to 

```C#
    ... 
<    issues = filteredIssues;
>    return filteredIssues;
}
else
{
    return issues.ToList();
}

Reply:
Discussed offline. 

What is happening is that for each Issue we have to traverse each single Label that the Issue has. We need to make sure that that label is not part of the list of labels that we have to exclude. We have to traverse the whole list of labels before we make a determination if that specific Issue should be excluded or not. Even if I used HashSet I will still need to keep the "skip" variable.

Reply:
Kevin is okay with keeping the current implementation